run-integration-test/action.yaml

```
@@ -314,7 +314,9 @@ runs:
 
     - name: Extract Failed Tests
       id: failed-tests
-      if: steps.integration-test.conclusion == 'failure'
+      # The success() function is automatically used if no other status function is used.
+      # See https://docs.github.com/en/actions/reference/workflows-and-actions/expressions#failure-with-conditions
+      if: ${{ failure() && steps.integration-test.conclusion == 'failure' }}
```

I still think you can avoid the extra interpolation syntax. `if` is special that way:

```suggestion
      if: && steps.integration-test.conclusion == 'failure'
```